core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -729,7 +729,14 @@ public String asString(@Nullable String decodingCharset) throws IOException {
 		// save the generated String as the request before returning it
 		// Specify initial capacity a little larger than file-size just as extra safeguard we do not re-allocate buffer.
 		String result = StreamUtil.readerToString(asReader(decodingCharset), null, false, (int) size() + 32);
-		if(!isBinary() || !isRepeatable()) {
+		if (!isBinary() || !isRepeatable()) {
+			if (request instanceof AutoCloseable) {
+				try {
+					((AutoCloseable) request).close();
+				} catch (Exception e) {
+					throw new IOException("could not close request of type [" + requestClass + "], inside message " + this, e);
```

Comment:
Is het beter om een exception in close door te gooien, of beter om deze te loggen en negeren?

Reply:
Ja, ik twijfelde er over, wat in deze situatie het beste is. Voorheen werd de `request` gewoon overschreven en gebeurde ook geen close. Misschien is het wel meer fail-save toch om het te loggen (als warning?) en negeren. Eens?
De `message.asString()` had trouwens dus altijd al het side-effect dat de request werd omgezet naar een String, als hij niet binair of niet repeatable is. Blijkbaar is het dan altijd de laatste handeling oid op een Message, wat men wil doen.

Reply:
Ik vind dit wel raar hoor. De reader krijgt een stream mee. Zodra je de reader sluit (in StreamUtil.readerToString) dan zou dat allemaal ge propageert moeten worden. In plaats van hier de close aan te roepen, kan je misschien iets meer vertellen over de stack en wie dan dus niet die close aanroept?

Reply:
Het is de situatie waar `request` wordt overschreven met de nieuwe string. Dus wat er in `request` zat, zou inderdaad gesloten moeten worden, anders wordt die `close()` nooit meer aangeroepen.
En omdat we de `PipeLineSession` hier niet hebben kunnen we ook niet schedulen voor `closeOnClose`.

Dus ja, het is raar en ik moest er ook even goed naar kijken, maar ik denk wel dat het klopt.

Reply:
Vandaag besproken samen.